.github/workflows/deploy.yml

```diff
@@ -0,0 +1,78 @@
+      - 'aurorawatch-pipeline/**'
+      - 'sms-checker/**'
+
+jobs:
+  build_and_push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up AWS CLI
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - name: Build and Push Dashboard Docker Image
+        if: ${{ hashFiles('dashboard/Dockerfile') != '' }}
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_DASHBOARD }}
+          docker build -t dashboard:latest ./dashboard --platform "linux/amd64"
+          docker tag dashboard:latest ${{ secrets.ECR_DASHBOARD }}:latest
+          docker push ${{ secrets.ECR_DASHBOARD }}:latest
+
+      - name: Build and Push Quadhoral-Openmeteo Docker Image
+        if: ${{ hashFiles('quadhoral-openmeteo/Dockerfile') != '' }}
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_QUADHORAL_OPENMETEO }}
+          docker build -t quadhoral-openmeteo:latest ./quadhoral-openmeteo --platform "linux/amd64"
+          docker tag quadhoral-openmeteo:latest ${{ secrets.ECR_QUADHORAL_OPENMETEO }}:latest
+          docker push ${{ secrets.ECR_QUADHORAL_OPENMETEO }}:latest
+
+      - name: Build and Push Weekly-Openmeteo Docker Image
+        if: ${{ hashFiles('weekly-openmeteo/Dockerfile') != '' }}
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_WEEKLY_OPENMETEO }}
+          docker build -t weekly-openmeteo:latest ./weekly-openmeteo --platform "linux/amd64"
+          docker tag weekly-openmeteo:latest ${{ secrets.ECR_WEEKLY_OPENMETEO }}:latest
+          docker push ${{ secrets.ECR_WEEKLY_OPENMETEO }}:latest
+
+      - name: Build and Push Astronomy Pipeline Docker Image
+        if: ${{ hashFiles('astronomy-pipeline/Dockerfile') != '' }}
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_ASTRONOMY_PIPELINE }}
+          docker build -t astronomy-pipeline:latest ./astronomy-pipeline --platform "linux/amd64"
+          docker tag astronomy-pipeline:latest ${{ secrets.ECR_ASTRONOMY_PIPELINE }}:latest
+          docker push ${{ secrets.ECR_ASTRONOMY_PIPELINE }}:latest
+
+      - name: Build and Push Aurorawatch Pipeline Docker Image
+        if: ${{ hashFiles('aurorawatch-pipeline/Dockerfile') != '' }}
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_AURORAWATCH_PIPELINE }}
```

Comment:
Good use of secrets throughout to stop certain people trying any funny business

Reply:
That would be seriously unfunny, dude.

Reply:
(Mehrdad)

Reply:
I have also set a cooldown where people who aren't direct collaborators of this repo cannot make issues